src/main/java/io/cdap/plugin/gcp/gcs/StorageClient.java

```diff
@@ -115,9 +115,67 @@ public void mapMetaDataForAllBlobs(String path, Consumer<Map<String, String>> fu
    * @param recursive whether to copy objects in all subdirectories
    * @param overwrite whether to overwrite existing objects
    * @throws IllegalArgumentException if overwrite is false and copying would overwrite an existing object
+   * @return copied blob object from google bucket storage or null if we are dealing with a directory/bucket
    */
-  public void copy(GCSPath sourcePath, GCSPath destPath, boolean recursive, boolean overwrite) {
+  public Blob copy(GCSPath sourcePath, GCSPath destPath, boolean recursive, boolean overwrite) {
+    //If sourcePath is not a bucket and sourcePath doesn't end with "/"
+    //For example: gs://plugin_376/csvexample.csv
+    //It means we are dealing with a single object which can be copied to the destination
+    if (!sourcePath.isBucket() && !sourcePath.getName().endsWith("/")) {
+      return copyObjectToDestination(sourcePath, destPath, overwrite);
+    }
     pairTraverse(sourcePath, destPath, recursive, overwrite, BlobPair::copy);
+    return null;
+  }
+
+  /**
+   * Copy a single object from the source path to the destination path.
+   *
+   * @param sourcePath the path to copy object from
+   * @param destPath   the path to copy object to
+   * @param overwrite  whether to overwrite existing object
+   * @throws RuntimeException if overwrite is false and copying would overwrite an existing object
+   * @throws StorageException if storage cannot be accessed or cannot be found
+   * @return copied blob object from google bucket storage
+   */
+  private Blob copyObjectToDestination(GCSPath sourcePath, GCSPath destPath, boolean overwrite) {
+    Blob sourceBlob;
+    try {
+      sourceBlob = storage.get(sourcePath.getBucket(), sourcePath.getName());
+      if (sourceBlob == null) {
+        throw new RuntimeException(
+          String.format("Object %s. not found", sourcePath.getUri())
+            + "Ensure you entered the correct path.");
+      }
+    } catch (StorageException e) {
+      throw new RuntimeException(
```

Comment:
Why are we wrapping StoroageException in RuntimeException? javadoc for this method indicates that it throws StorageException

Reply:
It's something related with this ticket: https://cdap.atlassian.net/browse/PLUGIN-202

Something similar was also done in _pairTraverse_ method: https://github.com/data-integrations/google-cloud/blob/69c37be020c1d33d6ee8ecf8d856161945530a34/src/main/java/io/cdap/plugin/gcp/gcs/StorageClient.java#L276-L283 ...and the reason for this was to catch some edge cases and to present more meaningful messages to the user.

Reply:
I see, then let's just fix the javadoc (@throws) as we are not throwing StorageException.